src/ceph.in

```diff
@@ -899,7 +899,10 @@ def main():
     try:
         if childargs and childargs[0] == 'ping':
             return ping_monitor(cluster_handle, childargs[1], timeout)
-        run_in_thread(cluster_handle.connect, timeout=timeout)
+        result = run_in_thread(cluster_handle.connect, timeout=timeout)
+        if type(result) is tuple and result[0] == -errno.EINTR:
```

Comment:
or...not.

So, it turns out that run_in_thread can return darn near anything, depending on what you're running. So, it is not particularly amenable to standardization. We could have it return a dict with an "error" item, but otherwise we are stuck with this approach.